### PR TITLE
Improve lock file management in Windows

### PR DESCRIPTION
**Pull request by diegopradogesto**
The Windows API is now used to create and delete the lock file. This file is now created with a flag DELETE_ON_CLOSE that deletes the file when closing or when the application gets interrupted.

### CODE REVIEW

File: UM/Application.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def __init__(self, name: str, version: str, build_type: str = "", is_debug_mode:
         self._is_headless = False #type: bool
         self._use_external_backend = False #type: bool
 
-        self.CONFIG_LOCK_FILENAME = "{name}.lock".format(name = self._app_name)
+        self.CONFIG_LOCK_FILENAME = "{name}.lock".format(name = self._app_name) # type: str
```
Comment (fieldOfView) [inline UM/Application.py]: Out of interest, why not ```self._config_lock_filename```? IE: why is this not a private variable like any other?

Comment (diegopradogesto) [reply UM/Application.py]: Oh, yes. I just copied as it was in the ContainerRegistry. It makes sense to change it.

**Comment by thopiekar** [comment]
Are you sure that this functionality is not delivered by the tempfile module?
I remember that there was a function which was supposed to create a temporary file, which gets removed after being closed.
This sounds pretty close to what you want and have. Don't know whether it acts the same.

**Comment by LipuFei** [comment]
@thopiekar temp file does automatically remove it when it's closed, but we need that file to be exclusively owned by the process that created it, thus the name "lock file".

**Comment by thopiekar** [comment]
Hmm, I still have something in mind that tells me that there are no real
lockfiles on Windows.
Therefore I thought using a tempfile would be good for Windows only.

But yeah, if that's now or if it was always supported on windows, then
excuse the noise on this 😅

**Pull request merged**